### ISSUE TITLE
Parsing fails on a localhost certificate

### ISSUE THREAD

**Issue by flockoffiles** [filom/ASN1Decoder#34]
A colleague of mine generated a certificate with the following sequence (also see text dump below):

```
SEQUENCE {
    SEQUENCE {
        INTEGER : 8 bytes
        SEQUENCE {
            OBJECTIDENTIFIER : 1.2.840.113549.1.1.11 (sha256WithRSAEncryption)
            NULL
        }
        SEQUENCE {
            SET {
                SEQUENCE {
                    OBJECTIDENTIFIER : 2.5.4.6 (countryName)
                    PRINTABLESTRING : NL
                }
            }
            SET {
                SEQUENCE {
                    OBJECTIDENTIFIER : 2.5.4.10 (organizationName)
                    UTF8STRING : development
                }
            }
            SET {
                SEQUENCE {
                    OBJECTIDENTIFIER : 2.5.4.3 (commonName)
                    UTF8STRING : localhost
                }
            }
        }
        SEQUENCE {
            UTCTIME : 2021-06-28 10:29:27 +0000
            UTCTIME : 2022-06-28 10:29:27 +0000
        }
        SEQUENCE {
            SET {
                SEQUENCE {
                    OBJECTIDENTIFIER : 2.5.4.6 (countryName)
                    PRINTABLESTRING : NL
                }
            }
            SET {
                SEQUENCE {
                    OBJECTIDENTIFIER : 2.5.4.10 (organizationName)
                    UTF8STRING : development
                }
            }
            SET {
                SEQUENCE {
                    OBJECTIDENTIFIER : 2.5.4.3 (commonName)
                    UTF8STRING : localhost
                }
            }
        }
        SEQUENCE {
            SEQUENCE {
                OBJECTIDENTIFIER : 1.2.840.113549.1.1.1 (rsaEncryption)
                NULL
            }
            BITSTRING : 270 bytes
        }
    }
    SEQUENCE {
        OBJECTIDENTIFIER : 1.2.840.113549.1.1.11 (sha256WithRSAEncryption)
        NULL
    }
    BITSTRING : 256 bytes
}


```

Text dump:
```
Certificate:
    Data:
        Version: 1 (0x0)
        Serial Number: 9809880311725596992 (0x8823b22b2580a140)
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=NL, O=development, CN=localhost
        Validity
            Not Before: Jun 28 10:29:27 2021 GMT
            Not After : Jun 28 10:29:27 2022 GMT
        Subject: C=NL, O=development, CN=localhost
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:b4:74:fb:17:a0:08:15:6b:84:f9:1a:97:5d:73:
                    2c:9b:f2:f0:59:aa:6c:67:30:41:91:22:7f:c5:6b:
                    24:14:f3:3b:10:49:11:5c:ca:05:45:c1:39:0d:1c:
                    6a:85:8d:92:51:57:39:e4:96:0c:08:84:c8:2f:52:
                    32:6a:f5:5e:e9:e0:e4:d5:c8:df:e0:88:b0:d3:d4:
                    e3:33:6b:db:9f:fb:19:23:e0:02:70:af:25:43:6f:
                    75:f7:ae:f9:dc:ea:a6:60:f2:8b:df:4b:d9:ec:d6:
                    06:f3:9f:f5:68:d0:5e:23:86:44:81:f0:fb:9f:21:
                    31:4f:0e:bd:d9:58:11:26:89:37:2c:18:89:92:37:
                    e6:9a:5d:84:74:60:53:78:6f:ff:c8:41:ae:bc:e9:
                    14:6e:ed:0a:3b:38:bf:79:ac:7f:8c:db:27:d9:5e:
                    40:04:49:db:95:52:14:98:43:5e:5a:82:90:78:ff:
                    35:80:16:5e:67:7b:6f:02:9d:1e:ae:e3:57:76:34:
                    1b:c6:3c:f7:48:e0:d8:3c:9c:18:df:5f:6c:c4:5e:
                    f2:fc:b7:91:44:e9:95:f9:90:30:e8:81:f9:ae:b2:
                    9a:ef:2a:a4:32:4e:68:35:54:89:db:c2:22:65:34:
                    72:a7:73:70:29:9b:53:a3:ea:b7:69:ef:ae:78:09:
                    99:7b
                Exponent: 65537 (0x10001)
    Signature Algorithm: sha256WithRSAEncryption
         41:36:2b:f3:f4:72:b7:19:4e:84:cb:9c:e3:eb:1a:4a:12:3a:
         b7:09:d5:0d:1c:ea:0f:32:2a:c1:a5:d9:f7:3d:3f:c9:4f:8a:
         87:2b:35:b1:16:ea:74:86:11:29:70:72:b2:03:8d:92:92:fa:
         80:ea:e5:2a:57:60:ac:68:d4:7d:b6:50:71:ce:ca:f5:57:fc:
         f6:de:1d:7a:8d:76:ab:4c:c3:a5:5f:29:82:80:5a:ef:1c:e8:
         6c:44:5e:6a:c7:2c:08:1f:7d:fc:a0:c8:5b:33:40:7e:37:48:
         6e:16:94:51:c7:99:62:f3:a6:c8:5a:99:a0:db:39:bc:da:05:
         56:b8:51:ac:70:56:9b:19:53:44:41:2f:3b:be:50:95:af:6f:
         df:2f:02:a1:7a:6b:f5:09:5c:3d:66:b9:07:53:70:4f:03:4f:
         2b:f0:a8:ba:7f:7d:01:ca:a8:98:be:ed:b1:0b:b6:bd:74:74:
         f3:f2:f0:4b:3c:93:b4:c5:22:b0:32:3e:03:73:4a:93:2a:d1:
         bd:ea:2f:a3:f8:45:1d:53:18:61:af:55:4a:65:d2:f9:81:eb:
         af:07:45:14:c7:f6:8d:22:d9:c3:6f:9d:d2:a8:bb:7c:c1:3d:
         20:0d:62:27:e2:93:43:83:23:8d:ba:c4:9f:48:11:dd:3a:3b:
         04:f2:8a:9f
```

When trying to obtain the publicKey from this certificate, parsing fails.

**Comment by flockoffiles** [comment]
Here is the PEM file for the certificate:

```
-----BEGIN CERTIFICATE-----
MIIC6jCCAdICCQCII7IrJYChQDANBgkqhkiG9w0BAQsFADA3MQswCQYDVQQGEwJO
TDEUMBIGA1UECgwLZGV2ZWxvcG1lbnQxEjAQBgNVBAMMCWxvY2FsaG9zdDAeFw0y
MTA2MjgxMDI5MjdaFw0yMjA2MjgxMDI5MjdaMDcxCzAJBgNVBAYTAk5MMRQwEgYD
VQQKDAtkZXZlbG9wbWVudDESMBAGA1UEAwwJbG9jYWxob3N0MIIBIjANBgkqhkiG
9w0BAQEFAAOCAQ8AMIIBCgKCAQEAtHT7F6AIFWuE+RqXXXMsm/LwWapsZzBBkSJ/
xWskFPM7EEkRXMoFRcE5DRxqhY2SUVc55JYMCITIL1IyavVe6eDk1cjf4Iiw09Tj
M2vbn/sZI+ACcK8lQ29196753OqmYPKL30vZ7NYG85/1aNBeI4ZEgfD7nyExTw69
2VgRJok3LBiJkjfmml2EdGBTeG//yEGuvOkUbu0KOzi/eax/jNsn2V5ABEnblVIU
mENeWoKQeP81gBZeZ3tvAp0eruNXdjQbxjz3SODYPJwY319sxF7y/LeRROmV+ZAw
6IH5rrKa7yqkMk5oNVSJ28IiZTRyp3NwKZtTo+q3ae+ueAmZewIDAQABMA0GCSqG
SIb3DQEBCwUAA4IBAQBBNivz9HK3GU6Ey5zj6xpKEjq3CdUNHOoPMirBpdn3PT/J
T4qHKzWxFup0hhEpcHKyA42SkvqA6uUqV2CsaNR9tlBxzsr1V/z23h16jXarTMOl
XymCgFrvHOhsRF5qxywIH338oMhbM0B+N0huFpRRx5li86bIWpmg2zm82gVWuFGs
cFabGVNEQS87vlCVr2/fLwKhemv1CVw9ZrkHU3BPA08r8Ki6f30ByqiYvu2xC7a9
dHTz8vBLPJO0xSKwMj4Dc0qTKtG96i+j+EUdUxhhr1VKZdL5geuvB0UUx/aNItnD
b53SqLt8wT0gDWIn4pNDgyONusSfSBHdOjsE8oqf
-----END CERTIFICATE-----
```

**Comment by filom** [comment]
fixed with 72e0e9e604750e4f527316bb3f004895c4832085

**Comment by flockoffiles** [comment]
Wow. That's super fast!
Appreciate it! :-)
